test/gateways/trexle_test.exs

```diff
@@ -54,6 +56,275 @@ defmodule Gringotts.Gateways.TrexleTest do
     description: "For our valued customer, Mr. Potter"
   ]
 
+  @invalid_amount_response ~s/{"error":"Payment failed","detail":"Amount must be at least 50 cents"}/
+  @invalid_card_response ~s/{"error":"Payment failed","detail":"Your card's expiration year is invalid."}/
+  @valid_request_response ~s/{"response":{"token":"charge_3e89c6f073606ac1efe62e76e22dd7885441dc72","success":true,"captured":false}}/
+  @invalid_token_response ~s/{"error":"Capture failed","detail":"invalid token"}/
+  @valid_token_response ~s/{"response":{"token":"#{@valid_token}","success":true,"captured":true,"amount":299,"status_message":"Transaction approved"}}/
+  @invalid_token_response_for_refund ~s/{"error":"Refund failed","detail":"invalid token"}/
+
+  describe "Store integration point testing" do
+    setup do
+      bypass = Bypass.open()
+      opts = @opts ++ [test_url: "http://localhost:#{bypass.port}/api/v1/"]
+      {:ok, bypass: bypass, opts: opts}
+    end
+
+    test "Store with valid card", %{bypass: bypass, opts: opts} do
+      Bypass.expect_once(bypass, "POST", "/api/v1/customers", fn conn ->
+        p_conn = parse(conn)
+        params = p_conn.body_params
+        assert params["email"] == @opts[:email]
+        assert params["ip_address"] == nil
+        assert params["description"] == nil
+        assert params["card"]["name"] == "#{@valid_card.first_name} #{@valid_card.last_name}"
+        assert params["card"]["number"] == @valid_card.number
+        assert params["card"]["address_line1"] == @address.street1
+        assert params["card"]["address_city"] == @address.city
+        Conn.resp(conn, 201, @invalid_token_response_for_refund)
+      end)
+
+      {:ok, response} = Trexle.store(@valid_card, opts)
+      assert response.status_code == 201
+    end
+  end
+
+  describe "Refund integration point testing" do
+    setup do
```

Comment:
This setup is common to all describe blocks, so you can keep it at the module level.

Reply:
Done